src/target/adiv5.c

```diff
@@ -294,9 +295,10 @@ static uint32_t adiv5_mem_read32(ADIv5_AP_t *ap, uint32_t addr)
 static uint32_t adiv5_ap_read_id(ADIv5_AP_t *ap, uint32_t addr)
 {
 	uint32_t res = 0;
+	uint8_t data[16];
```

Comment:
We're unsure with the merits and drawbacks of this change - could you put that in a separate PR as it needs some stringent testing etc and we'd like to merge the rest of the changeset.